src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -641,10 +638,31 @@ public void LoadEnvironmentVariables()
 
         public string LogMode => this.Log.Mode is null ? this.Defaults.LogMode : this.Log.Mode;
 
-        // TODO - Extensions
-        XmlNode? IWinSWConfiguration.ExtensionsConfiguration => throw new NotImplementedException();
+        public XmlNode? ExtensionsConfiguration => null;
+
+        // YAML Extension
+        [YamlMember(Alias = "extensions")]
+        public List<YamlExtensionConfiguration>? YamlExtensionsConfiguration { get; set; }
+
+        public List<string> ExtensionIds
+        {
+            get
+            {
+                if (this.YamlExtensionsConfiguration is null)
+                {
+                    return new List<string>(0);
+                }
+
+                var result = new List<string>();
```

Comment:
One of `List<T>` constructors accepts the capacity.

Reply:
why here need a capacity??

Reply:
The default capacity is 16.

Why do we need `ExtensionIds` at all?

Reply:
why does it matter since we foreach through the list? I am really not getting this point :(

Reply:
It slightly increases the memory consumption, because it is unlikely that we have 16+ extensions defined in the config. Autoscaling will be also inefficient. At the same time, none of that causes real issues with performance, so it is rather about code housekeeping

Reply:
Great! With that, I found this
**Constructs a List. The list is initially empty and has a capacity of zero. Upon adding the first element to the list the capacity is increased to 16, and then increased in multiples of two as required.**
It is a new thing for me!!!! Thank you so much both of you!!!!!